ale_linters/elm/make.vim

```diff
@@ -3,20 +3,25 @@
 
 function! ale_linters#elm#make#Handle(buffer, lines)
     let l:output = []
+    let l:temp_dir = has('win32') ? $TMP : $TMPDIR
     for l:line in a:lines
         if l:line[0] ==# '['
             let l:errors = json_decode(l:line)
 
             for l:error in l:errors
-                call add(l:output, {
-                \    'bufnr': a:buffer,
-                \    'lnum': l:error.region.start.line,
-                \    'vcol': 0,
-                \    'col': l:error.region.start.column,
-                \    'type': (l:error.type ==? 'error') ? 'E' : 'W',
-                \    'text': l:error.overview,
-                \    'nr': -1,
-                \})
+                " Check if file is from the temp directory.
+                " Filters out any errors not related to the buffer.
+                if l:error.file[0:len(l:temp_dir)-1] == l:temp_dir
```

Comment:
The linter didn't catch this one, but I wonder if you should use `==#` here or `==?`. One is case-sensitive, the other is case-insensitive. `==` is one or the other, depending on the user's Vim configuration.

It might make sense to run a case-sensitive path check for Unix systems, and a case-insensitive check for Windows computers.

I'd put some spaces around the binary `-` operator while you're at it.